test/e2e/specs/martech/tos-screenshots__checkout.ts

```diff
@@ -2,103 +2,66 @@
+		await sidebarComponent.navigate( 'Upgrades', 'Plans' );
+	} );
+
+	it( 'Add WordPress.com Business plan to cart', async function () {
+		const plansPage = new PlansPage( page );
+		await Promise.all( [ page.waitForURL( /.*checkout.*/ ), plansPage.selectPlan( 'Business' ) ] );
 	} );
 
-	describe( 'ToS screenshots of WP.com checkout in desktop, tablet, and mobile viewports', function () {
-		jest.setTimeout( 1800000 );
-		let cartCheckoutPage: CartCheckoutPage;
-		const magnificientNonEnLocales = [
-			'pt-br',
-			'fr',
-			'es',
-			'de',
-			'he',
-			'ja',
-			'it',
-			'nl',
-			'ru',
-			'tr',
-			'id',
-			'zh-cn',
-			'zh-tw',
-			'ko',
-			'ar',
-			'sv',
-		];
-
-		it( 'Login to marTech user account', async function () {
-			const loginPage = new LoginPage( page );
-			await loginPage.visit( { path: 'new' } );
-			const { username, password } = DataHelper.getAccountCredential( 'martechTosUser' );
-			await loginPage.logInWithCredentials( username, password );
-		} );
-
-		it( 'Set store cookie', async function () {
-			await BrowserManager.setStoreCookie( page, { currency: 'GBP' } );
-		} );
-
-		it( 'Set interface language to en', async function () {
-			await page.goto( DataHelper.getCalypsoURL( 'home' ), { waitUntil: 'networkidle' } );
-			const changeUILanguageFlow = new ChangeUILanguageFlow( page );
-			await changeUILanguageFlow.changeUILanguage( 'en' as LanguageSlug );
-			await page.goto( DataHelper.getCalypsoURL( 'home' ) );
-			await page.reload( { waitUntil: 'networkidle' } );
-		} );
-
-		it( 'Navigate to Upgrades > Plans', async function () {
-			const navbarCompnent = new NavbarComponent( page );
-			await navbarCompnent.clickMySites();
-			const sidebarComponent = new SidebarComponent( page );
-			await sidebarComponent.navigate( 'Upgrades', 'Plans' );
-		} );
-
-		it( 'Click on the "Plans" navigation tab', async function () {
-			plansPage = new PlansPage( page );
-			await plansPage.clickTab( 'Plans' );
-		} );
-
-		it( 'Click on "Upgrade" button for WordPress.com Business plan', async function () {
-			await plansPage.selectPlan( 'Business' );
-		} );
-
-		it( 'WordPress.com Business is added to cart', async function () {
-			cartCheckoutPage = new CartCheckoutPage( page );
-			await cartCheckoutPage.validateCartItem( cartItemForBusinessPlan );
-		} );
-
-		it( 'Screenshot checkout page for all en and non-en locales', async function () {
-			const changeUILanguageFlow = new ChangeUILanguageFlow( page );
-			for ( const locale of [ ...magnificientNonEnLocales, 'en' ] ) {
-				page.setViewportSize( { width: 1280, height: 720 } );
-				await page.goto( DataHelper.getCalypsoURL( 'home' ), { waitUntil: 'networkidle' } );
-				await changeUILanguageFlow.changeUILanguage( locale as LanguageSlug );
-				await page.waitForSelector( '.account__settings .language-picker__name-label' );
-				await page.goto( DataHelper.getCalypsoURL( 'home' ) );
+	describe.each( DataHelper.getMag16Locales() )(
```

Comment:
Key change here is that instead of using a `for` loop to change languages and capture screenshots, we use proper parametrization. This would offer several advantages:
- proper reporting of failures in case one of the MAG-16 languages cause an issue;
- use of `beforeAll` to set up the language switch